torchvision/prototype/transforms/functional/_misc.py

```diff
@@ -69,18 +69,18 @@ def normalize(
     return normalize_image_tensor(inpt, mean=mean, std=std, inplace=inplace)
 
 
-def _get_gaussian_kernel1d(kernel_size: int, sigma: float) -> torch.Tensor:
+def _get_gaussian_kernel1d(kernel_size: int, sigma: float, dtype: torch.dtype, device: torch.device) -> torch.Tensor:
     lim = (kernel_size - 1) / (2 * math.sqrt(2) * sigma)
-    x = torch.linspace(-lim, lim, steps=kernel_size)
+    x = torch.linspace(-lim, lim, steps=kernel_size, dtype=dtype, device=device)
```

Comment:
Following the approach of 2d, we should create directly the linspace on device/type rather than copying. Minor improvement for GPU:
```
[--------------------- Gaussian_blur cpu torch.uint8 ---------------------]
                     |  gaussian_blur v2 before  |  gaussian_blur_ v2 after
1 threads: ----------------------------------------------------------------
      (3, 400, 400)  |            4.0           |             3.97         
6 threads: ----------------------------------------------------------------
      (3, 400, 400)  |            5.3           |             5.28         

Times are in milliseconds (ms).

[--------------------- Gaussian_blur cuda torch.uint8 --------------------]
                     |  gaussian_blur v2 before  |  gaussian_blur_ v2 after
1 threads: ----------------------------------------------------------------
      (3, 400, 400)  |            165            |            153          
6 threads: ----------------------------------------------------------------
      (3, 400, 400)  |            165            |            154          

Times are in microseconds (us).

[-------------------- Gaussian_blur cpu torch.float32 --------------------]
                     |  gaussian_blur v2 before  |  gaussian_blur_ v2 after
1 threads: ----------------------------------------------------------------
      (3, 400, 400)  |            3.12           |            3.12         
6 threads: ----------------------------------------------------------------
      (3, 400, 400)  |            4.39           |            4.38         

Times are in milliseconds (ms).

[-------------------- Gaussian_blur cuda torch.float32 -------------------]
                     |  gaussian_blur v2 before  |  gaussian_blur_ v2 after
1 threads: ----------------------------------------------------------------
      (3, 400, 400)  |            134            |            122          
6 threads: ----------------------------------------------------------------
      (3, 400, 400)  |            135            |            122          

Times are in microseconds (us).
```

Reply:
Can `dtype` be an integer one here? Recall, that there were some differences in https://github.com/pytorch/vision/pull/6812#discussion_r1002089826.

Reply:
No it can only be float types. There is a conversion to float before the kernel is called:
https://github.com/pytorch/vision/blob/788ad12e5dcec1dffc7d6311e59a2be2792799b8/torchvision/prototype/transforms/functional/_misc.py#L126-L128